### PR TITLE
Support and fix merge layers (e.g. concat)

### PR DESCRIPTION
**Pull request by elad-c**
## Pull Request Description:
Support Keras Merge layers (Add, Concatenate, etc.) in model builder & quantization.
Fix for TF Merge functions (e.g. tf.concat).

## Checklist before requesting a review:
- [x] I set the appropriate labels on the pull request.
- [x] I have added/updated the release note draft (if necessary).
- [x] I have updated the documentation to reflect my changes (if necessary).
- [x] All function and files are well documented. 
- [x] All function and classes have type hints.
- [x] There is a licenses in all file.
- [x] The function and variable names are informative. 
- [x] I have checked for code duplications.
- [x] I have added new unittest (if necessary).

### CODE REVIEW

File: model_compression_toolkit/core/pytorch/back2framework/pytorch_model_builder.py [hunk 1]
```diff
@@ -139,7 +139,11 @@ def _run_operation(n: BaseNode,
         _tensor_input_allocs = None
 
     if isinstance(n, FunctionalNode) and n.inputs_as_list:
-        out_tensors_of_n_float = op_func(input_tensors, *op_call_args, **functional_kwargs)
+        if isinstance(op_func, PytorchQuantizationWrapper):
+            # in wrapped nodes, the op args & kwargs are already in the PytorchQuantizationWrapper.
+            out_tensors_of_n_float = op_func(*input_tensors)
```
Comment (reuvenperetz) [inline model_compression_toolkit/core/pytorch/back2framework/pytorch_model_builder.py:144]: Is there a scenario where op_func may be PytorchQuantizationWrapper?

File: model_compression_toolkit/core/pytorch/reader/graph_builders.py [hunk 2]
```diff
@@ -232,10 +232,19 @@ def nodes_builder(model: GraphModule,
...
+                for input_node in node.all_input_nodes:
+                    for i, input_arg in enumerate(node.args[0]):
+                        if input_node is input_arg and input_node in consts_dict:
+                            used_consts.add(input_node)
+                            weights.update({i: consts_dict[input_node]})
+            else:
+                for i, input_node in enumerate(node.all_input_nodes):
+                    if input_node in consts_dict:
+                        used_consts.add(input_node)
+                        weights.update({i: consts_dict[input_node]})
```
Comment (reuvenperetz) [inline model_compression_toolkit/core/pytorch/reader/graph_builders.py:247]: Consider extracting the common logic into a helper function to avoid duplication (not crucial).

**Review by reuvenperetz — APPROVED** [review]
